스케줄 타입 추가 및 사용자 권한 오류 수정 · 동료 스케줄 조회 로직 개선

junbem opened the pull request:
## 이슈 번호

close #

## 작업 사항

- ScheduleType에 RESTTIME("휴게시간") 추가
- refactor: 휴게가 근무로 표시되던 문제 수정
- WorkScheduleService#createBreakTimeSchedules에서 제목을 ScheduleType.RESTTIME.getDescription()으로 변경
- scheduleType을 WORK → RESTTIME으로 수정
- FE의 enum 우선 표기 로직과 일치하도록 정합성 개선

- 사용자 정보 조회 권한 제한 해결
  * UserController: @PreAuthorize 권한 조건 제거
  * 동료 정보 조회 시 발생하던 권한 평가 오류 해결

- 동료 스케줄 조회 서비스 개선 (WorkScheduleService)
  * Authorization 헤더 전달 방식으로 변경
  * 사용자 정보 추출 시 안전한 null 체크 추가
  * 부서 정보 필드명 수정 (departmentName)
  * 스케줄 조회 과정 상세 로깅 추가

- WorkScheduleController Authorization 헤더 파라미터 추가
- 사용자 정보 조회 권한 제한 해결
  * UserController: @PreAuthorize 권한 조건 제거
  * 동료 정보 조회 시 발생하던 권한 평가 오류 해결

- 동료 스케줄 조회 서비스 개선 (WorkScheduleService)
  * Authorization 헤더 전달 방식으로 변경
  * 사용자 정보 추출 시 안전한 null 체크 추가
  * 부서 정보 필드명 수정 (departmentName)
  * 스케줄 조회 과정 상세 로깅 추가

- WorkScheduleController Authorization 헤더 파라미터 추가

## 스크린샷 (선택)


## 공유사항

junbem committed:
feat(attendance): RESTTIME 스케줄 타입 추가 및 휴게시간 표시 개선
backend
ScheduleType에 RESTTIME("휴게시간") 추가
refactor: 휴게가 근무로 표시되던 문제 수정

junbem committed:
fix(attendance): 휴게 스케줄을 RESTTIME으로 저장하도록 변경
WorkScheduleService#createBreakTimeSchedules에서 제목을 ScheduleType.RESTTIME.getDescription()으로 변경
scheduleType을 WORK → RESTTIME으로 수정
FE의 enum 우선 표기 로직과 일치하도록 정합성 개선

junbem committed:
feat(attendance-service): 스케줄 타입/보안/시간표준 및 에러 메시지 개선
ScheduleType에 RESTTIME("휴게시간") 추가
WorkScheduleService
createWorkDaySchedules 제목을 ScheduleType.WORK.getDescription() 사용으로 변경(“근무일” 하드코딩 제거)
createBreakTimeSchedules를 RESTTIME으로 생성하도록 수정(제목/타입/색상 정비)
existsConflictingSchedule 호출 시 LocalTime → String 변환(requestDto.getStartTime().toString() 등)으로 시그니처 일치
create/update/delete 스케줄 실패 시 원인 메시지 전파(throw new RuntimeException(e.getMessage(), e))
WorkScheduleController
@PreAuthorize에서 authentication.principal.userId → authentication.principal.id로 수정
스케줄 생성 실패 시 상세 메시지 응답("스케줄 생성 실패: " + e.getMessage())
AttendanceController
@PreAuthorize의 userId 참조를 authentication.principal.id로 통일
쿼리 파라미터 LocalDateTime → Instant로 스펙 변경(수신 시 변환 제거)
AttendanceService(Interface/Impl)
checkIn/checkOut/markAttendanceStatus/markWorkStatus 등 파라미터 LocalDateTime → Instant로 전환
내부 계산은 ZonedDateTime(Asia/Seoul)로 처리하고 DB 저장은 Instant(UTC)로 통일
this week/auto checkout 등 시간 계산 로직 Instant.now() + ZoneId.of("Asia/Seoul") 사용
엔티티 시간 필드 일괄 전환(LocalDateTime → Instant)
Schedule, WorkPolicy, AnnualLeave, WorkTimeAdjustment, LeaveRequest 등 createdAt/updatedAt(및 requestedAt/approvedAt) Instant로 변경
엔티티 라이프사이클 훅을 Instant.now()로 변경
DB 마이그레이션
V20250904instant_utc_postgres.sql 추가: 주요 시간 컬럼을 timestamptz(UTC)로 전환(기존 데이터 Asia/Seoul 가정 변환 포함)
로깅/검증
사용자 권한 체크 로직 개선(본인 또는 ADMIN)
고정/정책 기반 스케줄 생성 시 로깅 강화

junbem committed:
Merge branch 'develop' into feature/workpolicy-api-integration

junbem committed:
Merge branch 'develop' into feature/workpolicy-api-integration

junbem committed:
Merge branch 'develop' into feature/workpolicy-api-integration

junbem committed:
Merge branch 'develop' into feature/workpolicy-api-integration

junbem committed:
fix(BE): 동료 근무표 조회 권한 오류 및 안정성 개선

- 사용자 정보 조회 권한 제한 해결
  * UserController: @PreAuthorize 권한 조건 제거
  * 동료 정보 조회 시 발생하던 권한 평가 오류 해결

- 동료 스케줄 조회 서비스 개선 (WorkScheduleService)
  * Authorization 헤더 전달 방식으로 변경
  * 사용자 정보 추출 시 안전한 null 체크 추가
  * 부서 정보 필드명 수정 (departmentName)
  * 스케줄 조회 과정 상세 로깅 추가

- WorkScheduleController Authorization 헤더 파라미터 추가

수정 파일:
- attendance-service/WorkScheduleController.java
- attendance-service/WorkScheduleService.java

junbem committed:
Merge branch 'develop' into feature/workpolicy-api-integration

junbem committed:
Merge branch 'develop' into feature/workpolicy-api-integration

hyeblee commented on attendance-service/src/main/java/com/hermes/attendanceservice/controller/WorkScheduleController.java:
```diff
@@ -259,16 +259,11 @@ public ResponseEntity<ApiResult<ColleagueScheduleResponseDto>> getColleagueSched
             @RequestParam @DateTimeFormat(iso = DateTimeFormat.ISO.DATE) LocalDate startDate,
             @RequestParam @DateTimeFormat(iso = DateTimeFormat.ISO.DATE) LocalDate endDate,
-            @AuthenticationPrincipal UserPrincipal user) {
+            @RequestHeader(value = "Authorization", required = false) String authorization) {
```
여기서 @AuthenticationPrincipal에서  인증 헤더를 직접 꺼내는 @RequestHeader("Authorization")로 바꾸신 이유가 있을까요?

junbem replied:
해당 부분 수정했습니다

hyeblee replied:
확인했습니다!

junbem committed:
refactor: getUserWorkPolicy 메서드 호출 파라미터 정리 완료

- 컨트롤러와 서비스 전체에서 불필요한 authorization/null 파라미터 제거
- WorkScheduleController.getUserWorkPolicy 호출 부분 수정
- WorkScheduleService 내부의 모든 getUserWorkPolicy 호출 정리
- 메서드 시그니처 일관성 및 코드 간소화 달성
